### ISSUE TITLE
After Update 1.17.0 not working anymore

### ISSUE THREAD

**Issue by walhallaRV** [wallabag/wallabagger#391]
Hi,
doesn't save bookmarks anymore. If I try to wallabag a site, there comes "Saving the site in wallabag" and after some seconds it appears "An error occurred while saving ...". 

The server is runnig without any problems. I can save bookmarks manually. No errors.

I deleted the profile in the addon and recreated. Same result. 

I deleted the client on the server and recreated, created a new profile in the addon. Same result. 

If I check the options / configuration of the addon, all the checks are "green" or OK. No errors there. 

AddOn (Chrome): V1.17.0
Wallabag Server: V2.6.9

Thanks for any hint to resolve this problem!

![ScreenShot 859 Wallabagger - Vivaldi](https://github.com/wallabag/wallabagger/assets/27780122/342a6989-ba9a-42d8-8727-9bb40e6fbdf8)

**Comment by Simounet** [comment]
Hi @walhallaRV , I'm really sorry for your trouble. Just to be sure, are you using Vivaldi? I want to try to reproduce your issue.

**Comment by walhallaRV** [comment]
Hi,
I made the screenshot with Vivaldi, yes. But I also tried with Chrome and Opera -> exactly the same behaviour / error. 

Thanks for your response!

**Comment by Simounet** [comment]
I'm assuming that you are using a self hosted version of wallabag.
- Did you updated it recently?
- Was it working before?
- Are you sure it's not working after the wallabagger update?
- Can you enable the debug mode at the end of the option page, then try to save a page?Then give me the error result that should appear on the extension page (`chrome://extensions/`)?

**Comment by walhallaRV** [comment]
@  you are using a self hosted version of wallabag
-> yes

@ Did you updated it recently?
-> no. some weeks ago. but worked. any hint regarding this or known problems? 

@ Was it working before?
-> yes

@ Are you sure it's not working after the wallabagger update?
hmmm,, pretty sure yes, but can not bet ;)

@ debug: will try later when i am at home!

thanks so far!

**Comment by walhallaRV** [comment]
"Then give me the error result that should appear on the extension page"
-> I tryed, but I cant see anything on the extension page (neighter Vivaldi, nor Chrome)

Just can see the list of AddOns installed. On "Details" dont see anything too?

**Comment by Simounet** [comment]
Don't you have `DELETE` and `ERRORS` buttons after the `DETAILS` button?

**Comment by walhallaRV** [comment]
"DELETE" I have, yes. "Errors" no ->

![ScreenShot 871 Erweiterungen - Vivaldi](https://github.com/wallabag/wallabagger/assets/27780122/0de34789-c90f-4c0c-a782-5f91a3354e0c)

**Comment by Simounet** [comment]
Ok, can you click on `Hintergrundseite` and let me know what message you should have on the bottom part after trying to save a page please?

**Comment by walhallaRV** [comment]
Hi again, "401" OK. Used my notebook to test now, deleted Chrome caches, history, cookies, ... everything, restarted Chrome, flushed local DNS-cache on the notebook and tried again. Maybe some more information for you ->

![ScreenShot 893 DevTools - chrome-extension___gbmgphmejlcoihgedabhgjdkcahacjlj__generated_background_page html](https://github.com/wallabag/wallabagger/assets/27780122/c12876f2-b8d7-4896-8b37-f28cd2438d28)

**Comment by Simounet** [comment]
It's so weird. It looks like you're not authenticated with your wallabag instance. I'm trying to reproduce this behavior on my side. Do you have any addon that can modify the request? Can you try on a clean browser profile? Firefox maybe?

**Comment by walhallaRV** [comment]
This Chrome is nearly fresh and clean. Just two AddOns: Bitwarden, VPN-Provider and yours ;). I deactivated the other two - same error.

The user / password are correct. I checked once again. When I created this Profile in your AddOn, I put in exactly this data and the result was "OK" / green (see screenshot in my first post -> "Checklist"). To be safe, I logged in directly in Wallabag and copied the data from the error -> everything OK, login succeeded.

If it helps, I can create an user for you in Wallabag. So you can see / try what happens on your computer?

**Comment by HolgerAusB** [comment]
Just wild guessing, because I had a similar problem with FieryFeeds connecting to my Wallabag.

I changed the lifetime for tokens in `[wallabag]/app/config/parameters.yml` to a much high value (months instead of  1 hour)

```
    # how long the access token should live in seconds for the API
    fos_oauth_server_access_token_lifetime: 3600
    # how long the refresh token should life in seconds for the API
    fos_oauth_server_refresh_token_lifetime: 1209600
```

After changing this back to standard, FieryFeeds still tried to use it's stored access token, because it thought it is valid for the next few months. So I needed to generate a new wallabag API key for FieryFeeds.

And as you exposed your client_id and client_secret to us, you should revoke this an generate a new key set anyway.

**Comment by Simounet** [comment]
> This Chrome is nearly fresh and clean. Just two AddOns: Bitwarden, VPN-Provider and yours ;). I deactivated the other two - same error.
> 
> The user / password are correct. I checked once again. When I created this Profile in your AddOn, I put in exactly this data and the result was "OK" / green (see screenshot in my first post -> "Checklist"). To be safe, I logged in directly in Wallabag and copied the data from the error -> everything OK, login succeeded.
> 
> If it helps, I can create an user for you in Wallabag. So you can see / try what happens on your computer?

Would be very helpful. If you want, you can send me the credentials to `contact at mygithubusername dot net`.

**Comment by walhallaRV** [comment]
> Just wild guessing, because I had a similar problem with FieryFeeds connecting to my Wallabag.
> 
> I changed the lifetime for tokens in `[wallabag]/app/config/parameters.yml` to a much high value (months instead of 1 hour)
> 
> ```
>     # how long the access token should live in seconds for the API
>     fos_oauth_server_access_token_lifetime: 3600
>     # how long the refresh token should life in seconds for the API
>     fos_oauth_server_refresh_token_lifetime: 1209600
> ```
> 
> After changing this back to standard, FieryFeeds still tried to use it's stored access token, because it thought it is valid for the next few months. So I needed to generate a new wallabag API key for FieryFeeds.
> 
> And as you exposed your client_id and client_secret to us, you should revoke this and generate a new key set anyway.

Hi Holger,
i had a look in the parameters.yml ->

    fos_oauth_server_access_token_lifetime: 3600
    fos_oauth_server_refresh_token_lifetime: 1209600

I never changed it. Seems to be "standard". So what did you set here that it worked again?

@ client id / pw: yes, I know. As long as nobody knows where the waalabag is, I dont mind. Created several times new because it didnt work and sure: when it works again i will recreate. THANKS for the hint! ;)

**Comment by walhallaRV** [comment]
> > This Chrome is nearly fresh and clean. Just two AddOns: Bitwarden, VPN-Provider and yours ;). I deactivated the other two - same error.
> > The user / password are correct. I checked once again. When I created this Profile in your AddOn, I put in exactly this data and the result was "OK" / green (see screenshot in my first post -> "Checklist"). To be safe, I logged in directly in Wallabag and copied the data from the error -> everything OK, login succeeded.
> > If it helps, I can create an user for you in Wallabag. So you can see / try what happens on your computer?
> 
> Would be very helpful. If you want, you can send me the credentials to `contact at mygithubusername dot net`.

Sent you a test - email. If it was right, pls answer and I will send you the data! THX

**Comment by HolgerAusB** [comment]
I set it back to this standard. I just want to make sure, you haven't changed here anything.

**Comment by Simounet** [comment]
I can reproduce your error. Seems like a wallabag server error because with the exact same extension, I can use wallabagger without any issue with my instance and wallabag.it.
Can you try to use the wallabag app on Android/iPhone, just to be sure?

**Comment by HolgerAusB** [comment]
401 = unauthorized

- of course, you tried to generate a new API key_pair in wallabag?
- your are using the correct password? I went crazy the first time, when accidentally using the API of my non-privileged user with admin's password
- are you using two-factor-auth in wallabag? Don't know if this makes a difference here.

**Comment by Simounet** [comment]
@HolgerAusB Thanks for your help. It's not that kind of issue. @walhallaRV sent me credentials to test it and I can reproduce it.

**Comment by walhallaRV** [comment]
> @HolgerAusB Thanks for your help. It's not that kind of issue. @walhallaRV sent me credentials to test it and I can reproduce it.

I sent by email and saw that you logged in?!

**Comment by walhallaRV** [comment]
> 401 = unauthorized
> 
> * of course, you tried to generate a new API key_pair in wallabag?
> * your are using the correct password? I went crazy the first time, when accidentally using the API of my non-privileged user with admin's password
> * are you using two-factor-auth in wallabag? Don't know if this makes a difference here.

"you tried to generate a new API key_pair in wallabag?"
-> Yes, i did.

"your are using the correct password?"
-> Yes. I doublechecked several times. Identical with the password in the error - message. And all the tests went OK when I created the user - profile in the AddOn (see screenshot in my firts post).

"are you using two-factor-auth in wallabag?"
-> No, i don't.

Unfortunately 401 can mean many, many thing. But I cant find any details. Searched the logfile and just found "401" when the AddOn calls. Nothing more.

**Comment by HolgerAusB** [comment]
maybe something with site-config of webserver (apache2)?

**Comment by walhallaRV** [comment]
> I can reproduce your error. Seems like a wallabag server error because with the exact same extension, I can use wallabagger without any issue with my instance and wallabag.it. Can you try to use the wallabag app on Android/iPhone, just to be sure?

Hi,
I deleted all API-keys and changed my password. Deleted the profil in the AddOn too.

@ Android: made a connection check and ended in:

![Screenshot_2024-04-19-13-07-12-893_fr gaulupeau apps InThePoche](https://github.com/wallabag/wallabagger/assets/27780122/7279ae40-421d-467e-b0a5-43454c563731)

OAuth2??? 

@ AddOn
Created a new profile with the new data. It worked, everything green or "passed" ->

![ScreenShot 908 Wallabagger - Vivaldi](https://github.com/wallabag/wallabagger/assets/27780122/1bf94795-8fda-4185-b91e-0acff8f7d239)

If I try to save a website, appears "Loading API ..." (takes some seconds) -> "Saving website in wallabag"  -> "An error occurred while saving ..."

![ScreenShot 909 How to configure docker container with https - DevOpsSchool com - Vivaldi](https://github.com/wallabag/wallabagger/assets/27780122/cc6dae1a-2181-43eb-b0da-1e27e18c05c3)

![ScreenShot 910 Screenshots](https://github.com/wallabag/wallabagger/assets/27780122/16caf891-384f-4fe3-a4a3-94d710d9e1d4)

@ Wallabag (Web)
No problems. Can login, can save websites. Everything seems to work properly. No errors, clean logfile. 

So, I dont have any idea anymore. Seems that I have to give up ... :(

**Comment by Simounet** [comment]
Looks like this issue https://github.com/wallabag/wallabag/issues/2768 . Can you try this advice https://github.com/wallabag/wallabag/issues/2768#issuecomment-429399785?

**Comment by walhallaRV** [comment]
> maybe something with site-config of webserver (apache2)? apache-logfiles?

@ Logfiles clean. Just this simple lines with the 401 errors. Nothing else. php-Log is clean / empty.

@ site-config: didn't change, worked, but I can post it later from home!

**Comment by walhallaRV** [comment]
> Looks like this issue [wallabag/wallabag#2768](https://github.com/wallabag/wallabag/issues/2768) . Can you try this advice [wallabag/wallabag#2768 (comment)](https://github.com/wallabag/wallabag/issues/2768#issuecomment-426084967)?

Hi, 
I dont know what to say! read this two threads. 

Yes I have had this "CGIPassAuth on" in site-cfg: 

```
    <Directory /home/xxxxx/public_html/web>
    Require all granted
    CGIPassAuth On
       <IfModule mod_rewrite.c>
            Options +SymLinksIfOwnerMatch
            Options -MultiViews
            RewriteEngine On
            RewriteCond %{REQUEST_FILENAME} !-f
            RewriteRule ^(.*)$ app.php [QSA,L]
        </IfModule>
    </Directory>
```

Now I added this mentioned (in the thread): "adding SetEnvIf Authorization "(.*)" HTTP_AUTHORIZATION=$1 also solves the problem for me (apache2 version 2.4.10)." (mine is 2.4.59) which should be depreciated (not written in the man) 

**BUT IT WORKS NOW!** 

I don't understand, but OK ... I dont have to understand everything ;) Interesting is the fact that it worked! Maybe something changed with some of the PHP(fpm) or Apache Updates in the last days / weeks? We never will know. And unfortunately I cant say if it happened immediately after one of the updates (Debian / PHP / Apache)? Even worse: I realized that something doesnt work after I made the update of the AddOn (which I saw by the way, accidentially and made it ;). 

I am really sorry for making this trouble, robbing your time!!! I Iread hours and hours - buth at the wrong end. Didnt find / read this thread you sent. Maybe because I thought "Server was running. No problems. Worked." and for me the AddOn-Update was the evil :)))))

But OK. THANK YOU and Holger for the help and your hints / tipps! Maybe one day I can give a little bit back to you! ;)

THANKS, have a nice weekend, take care of you (litle bit ;) and enjoy life. 

**A la vôtre (? Google said ;) and Prost und zum Wohl!**
